In [ ]:
import pandas as pd
import sqlite3 as sql

In [ ]:
conn= sql.connect("/content/sql-murder-mystery.db")

Trial check Do not consider the below

In [ ]:
query="""
SELECT name, g.membership_id,g.check_in_date,g.check_in_time,g.check_out_time
FROM get_fit_now_member
JOIN (
    SELECT *
    FROM get_fit_now_check_in
    WHERE membership_id='90081'
) g ON get_fit_now_member.id=g.membership_id

"""

pd.set_option('display.max_colwidth',None)

pd.read_sql_query(query,conn)

name membership_id  check_in_date  check_in_time  check_out_time
0  Annabel Miller         90081       20180109           1600            1700

Display all the murder cases from the date Jan 15 2018 in the SQL City


In [ ]:
murders="""
SELECT *
FROM crime_scene_report
WHERE date='20180115'
AND city='SQL City'
AND type='murder';
"""

pd.set_option('display.max_colwidth',None)

pd.read_sql_query(murders,conn)

date    type  \
0  20180115  murder   

                                                                                                                                                                                 description  \
0  Security footage shows that there were 2 witnesses. The first witness lives at the last house on "Northwestern Dr". The second witness, named Annabel, lives somewhere on "Franklin Ave".   

       city  
0  SQL City

In [ ]:
murders="""
SELECT *
FROM crime_scene_report
WHERE date='20180115'
AND city='SQL City'
AND type='murder';
"""

pd.set_option('display.max_colwidth',None)

pd.read_sql_query(murders,conn)

WITNESS 1 details The person living in the last house of Northwestern Dr

Have ordered by address_number in descending order and limited to one


OUTPUT

    person_id	name	address	license_id	ssn
    14887	Morty Schapiro	4919 Northwestern Dr	118009	111564949

In [ ]:
witness_1_details = """
SELECT id as person_id,name, address_number||' '||address_street_name AS address,license_id	,ssn
FROM person
Where address_street_name='Northwestern Dr'
ORDER BY address_number DESC
LIMIT 1

"""
pd.set_option('display.max_colwidth',None)

pd.read_sql_query(witness_1_details,conn)

person_id            name               address  license_id        ssn
0      14887  Morty Schapiro  4919 Northwestern Dr      118009  111564949

WITNESS Details two Annabel from Franklin Ave Street

    person_id	name	address	license_id	ssn
	16371	Annabel Miller	103 Franklin Ave	490173	318771143

In [ ]:
witness_2_details="""
SELECT id as person_id,name, address_number||' '||address_street_name AS address,license_id	,ssn
FROM person
Where address_street_name='Franklin Ave'
AND name LIKE 'Annabel%';
"""

pd.set_option('display.max_colwidth',None)

pd.read_sql_query(witness_2_details,conn)

person_id            name           address  license_id        ssn
0      16371  Annabel Miller  103 Franklin Ave      490173  318771143

Witness interviews

Interview of witness 1



In [ ]:
witness_1_interviewdetails = """
SELECT id as person_id,name, address_number||' '||address_street_name AS address,license_id	,ssn, i.transcript
FROM person
JOIN (
    SELECT person_id, transcript
    From interview
) i ON person.id=i.person_id
Where address_street_name='Northwestern Dr'
ORDER BY address_number DESC
LIMIT 1

"""
pd.set_option('display.max_colwidth',None)

pd.read_sql_query(witness_1_interviewdetails,conn)

person_id            name               address  license_id        ssn  \
0      14887  Morty Schapiro  4919 Northwestern Dr      118009  111564949   

                                                                                                                                                                                                                        transcript  
0  I heard a gunshot and then saw a man run out. He had a "Get Fit Now Gym" bag. The membership number on the bag started with "48Z". Only gold members have those bags. The man got into a car with a plate that included "H42W".

In [ ]:
witness_2_interviewdetails="""
SELECT id as person_id,name, address_number||' '||address_street_name AS address,license_id	,ssn,i.transcript
FROM person
JOIN (
    SELECT person_id, transcript
    From interview
) i ON person.id=i.person_id
Where address_street_name='Franklin Ave'
AND name LIKE 'Annabel%';
"""

pd.set_option('display.max_colwidth',None)

pd.read_sql_query(witness_2_interviewdetails,conn)

person_id            name           address  license_id        ssn  \
0      16371  Annabel Miller  103 Franklin Ave      490173  318771143   

                                                                                                              transcript  
0  I saw the murder happen, and I recognized the killer from my gym when I was working out last week on January the 9th.

Checking gym database

FIT NOW Golden members with membership number starting with "48Z" details as specified by Witness 1


    id	person_id	name	membership_start_date	membership_status
	48Z7A	28819	Joe Germuska	20160305	gold
	48Z55	67318	Jeremy Bowers	20160101	gold

In [ ]:
gym_database="""
SELECT *
FROM get_fit_now_member
WHERE membership_status='gold'
AND id LIKE '48Z%'
"""

pd.set_option('display.max_colwidth',None)

pd.read_sql_query(gym_database,conn)

id  person_id           name  membership_start_date membership_status
0  48Z7A      28819   Joe Germuska               20160305              gold
1  48Z55      67318  Jeremy Bowers               20160101              gold

The data is shorted to two people JOE Germuska and JEREMY Bowers To still contract the data to one person querying based on witness two transcript

"last week on January the 9th"

In [ ]:
gym_database="""
SELECT *
FROM get_fit_now_member
JOIN(
    SELECT *
    FROM get_fit_now_check_in
    WHERE check_in_date LIKE 20180109
)
WHERE membership_status='gold'
AND id LIKE '48Z%'

"""

pd.set_option('display.max_colwidth',None)

pd.read_sql_query(gym_database,conn)

id  person_id           name  membership_start_date membership_status  \
0   48Z55      67318  Jeremy Bowers               20160101              gold   
1   48Z7A      28819   Joe Germuska               20160305              gold   
2   48Z55      67318  Jeremy Bowers               20160101              gold   
3   48Z7A      28819   Joe Germuska               20160305              gold   
4   48Z55      67318  Jeremy Bowers               20160101              gold   
5   48Z7A      28819   Joe Germuska               20160305              gold   
6   48Z55      67318  Jeremy Bowers               20160101              gold   
7   48Z7A      28819   Joe Germuska               20160305              gold   
8   48Z55      67318  Jeremy Bowers               20160101              gold   
9   48Z7A      28819   Joe Germuska               20160305              gold   
10  48Z55      67318  Jeremy Bowers               20160101              gold   
11  48Z7A      28819   Joe Germuska               20160305              gold   
12  48Z55      67318  Jeremy Bowers               20160101              gold   
13  48Z7A      28819   Joe Germuska               20160305              gold   
14  48Z55      67318  Jeremy Bowers               20160101              gold   
15  48Z7A      28819   Joe Germuska               20160305              gold   
16  48Z55      67318  Jeremy Bowers               20160101              gold   
17  48Z7A      28819   Joe Germuska               20160305              gold   
18  48Z55      67318  Jeremy Bowers               20160101              gold   
19  48Z7A      28819   Joe Germuska               20160305              gold   

   membership_id  check_in_date  check_in_time  check_out_time  
0          X0643       20180109            957            1164  
1          X0643       20180109            957            1164  
2          UK1F2       20180109            344             518  
3          UK1F2       20180109            344             518  
4          XTE42       20180109            486            1124  
5          XTE42       20180109            486            1124  
6          1AE2H       20180109            461             944  
7          1AE2H       20180109            461             944  
8          6LSTG       20180109            399             515  
9          6LSTG       20180109            399             515  
10         7MWHJ       20180109            273             885  
11         7MWHJ       20180109            273             885  
12         GE5Q8       20180109            367             959  
13         GE5Q8       20180109            367             959  
14         48Z7A       20180109           1600            1730  
15         48Z7A       20180109           1600            1730  
16         48Z55       20180109           1530            1700  
17         48Z55       20180109           1530            1700  
18         90081       20180109           1600            1700  
19         90081       20180109           1600            1700

Since there are a lot of time they have checked in on January 9th Checking the checkin and check out time of Annabelle Miller with id 90081


In [ ]:
gym_database="""
SELECT *
FROM get_fit_now_member
JOIN(
    SELECT *
    FROM get_fit_now_check_in
    WHERE check_in_date LIKE 20180109
    AND check_in_time='1600'
    AND	check_out_time ='1700'
)
WHERE membership_status='gold'
AND id LIKE '48Z%'

"""

pd.set_option('display.max_colwidth',None)

pd.read_sql_query(gym_database,conn)

id  person_id           name  membership_start_date membership_status  \
0  48Z7A      28819   Joe Germuska               20160305              gold   
1  48Z55      67318  Jeremy Bowers               20160101              gold   

  membership_id  check_in_date  check_in_time  check_out_time  
0         90081       20180109           1600            1700  
1         90081       20180109           1600            1700

Getting car detials based on witness two details


In [ ]:
query="""
SELECT name,license_id, d.plate_number
FROM person
JOIN(
SELECT *
FROM drivers_license
WHERE plate_number LIKE '%H42W%'
) d ON d.id=person.license_id
"""

pd.set_option('display.max_colwidth',None)

pd.read_sql_query(query,conn)

name  license_id plate_number
0  Tushar Chandra      664760       4H42WR
1   Jeremy Bowers      423327       0H42W2
2  Maxine Whitely      183779       H42W0X

SINCE only Jeremy Bowers name matches the plate no that includes 'H42' and thecheck in and check out time matches with the witness 2 and also based on statements of witness 1 I conclude that **JEREMY BOWERS as the KILLER **

Details of THE KILLER "JEREMY BOWERS"

In [ ]:
killer="""
SELECT name
FROM person
WHERE name="Jeremy Bowers"
"""

pd.set_option('display.max_colwidth',None)

pd.read_sql_query(killer,conn)

name
0  Jeremy Bowers

Interview statement of Jeremy Bowers


In [ ]:
killer_transacript="""
SELECT *
FROM person
"""

pd.set_option('display.max_colwidth',None)

pd.read_sql_query(killer_transacript,conn)

id                 name  license_id  address_number  \
0      10000   Christoper Peteuil      993845             624   
1      10007  Kourtney Calderwood      861794            2791   
2      10010            Muoi Cary      385336             741   
3      10016          Era Moselle      431897            1987   
4      10025         Trena Hornby      550890             276   
...      ...                  ...         ...             ...   
10006  99936          Luba Benser      274427             680   
10007  99941      Roxana Mckimley      975942            1613   
10008  99965      Cherie Zeimantz      287627            3661   
10009  99982          Allen Cruse      251350            3126   
10010  99990         Vance Hunten      830407            3056   

      address_street_name        ssn  
0            Bankhall Ave  747714076  
1           Gustavus Blvd  477972044  
2         Northwestern Dr  828638512  
3           Wood Glade St  614621061  
4           Daws Hill Way  223877684  
...                   ...        ...  
10006        Carnage Blvd  685095054  
10007             Gate St  512136801  
10008       The Water Ave  362877324  
10009           N Jean Dr  348734531  
10010       Lancefield St  896677562  

[10011 rows x 6 columns]

Based on his statement it seems tha he has been hired by someoneelse
Details
Height 5'5 to 5'7
Red Hair
Drives Tesla model S
Attended SLA Symphony Concert


In [ ]:
event_check="""
SELECT *
FROM facebook_event_checkin
JOIN(
    SELECT name,id
    FROM person
    JOIN (
        SELECT id as license_id FROM
        drivers_license
        WHERE car_make = "Tesla"
        AND car_model = "Model S"
        AND gender = "female"
        AND hair_color = "red"
    )d ON d.license_id=person.license_id
)p ON p.id=facebook_event_checkin.person_id
Where event_name="SQL Symphony Concert"
"""

pd.set_option('display.max_colwidth',None)

pd.read_sql_query(event_check,conn)


person_id  event_id            event_name      date              name  \
0      99716      1143  SQL Symphony Concert  20171206  Miranda Priestly   
1      99716      1143  SQL Symphony Concert  20171212  Miranda Priestly   
2      99716      1143  SQL Symphony Concert  20171229  Miranda Priestly   

      id  
0  99716  
1  99716  
2  99716


    person_id	event_id	event_name	date	name	id
    99716	1143	SQL Symphony Concert	20171206	Miranda Priestly	99716
	99716	1143	SQL Symphony Concert	20171212	Miranda Priestly	99716
	99716	1143	SQL Symphony Concert	20171229	Miranda Priestly	99716

All three records on differenct dates of the concert show the same person MIRANDA PRIESLY thus I conclude that the person hired the murderer is

***MIRANDA PRIESTLY ***